src/BlTools.MssqlFluentSqlWrapper/FluentSqlCommand.cs

```diff
@@ -0,0 +1,520 @@
+                                adapter.Fill(result.FillDataSetResult);
+                                break;
+                            }
+                            case ExecType.XmlReader:
+                            {
+                                using var reader = _command.ExecuteXmlReader();
+                                if (reader.Read())
+                                {
+                                    result.XmlResult = reader.ReadOuterXml();
+                                }
+
+                                break;
+                            }
+                        }
+                        
+                        Status = ExecStatus.Success;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    HandleException(ex);
+                }
+            }
+
+            return result;
+        }
+
+        private async Task<ExecResult> ExecAsync(ExecType execType, ExecResult result = null)
+        {
+            result ??= new ExecResult();
+            using (_leaveConnectionOpened ? null : _command.Connection) // using null is safe.
+            {
+                try
+                {
+                    if (_command.Connection.State != ConnectionState.Open)
+                    {
+                        await _command.Connection.OpenAsync();
+                    }
+
+                    using (_command)
+                    {
+                        switch (execType)
+                        {
+                            case ExecType.NonQuery:
+                            {
+                                result.NonQueryResult = await _command.ExecuteNonQueryAsync();
+                                break;
+                            }
+                            case ExecType.Scalar:
+                            {
+                                result.ScalarResult = await _command.ExecuteScalarAsync();
+                                break;
+                            }
+                            case ExecType.Reader:
+                            {
+                                using var reader = await _command.ExecuteReaderAsync();
+                                var resultNumber = 0;
+                                do
+                                {
+                                    while (await reader.ReadAsync())
+                                    {
+                                        _dataReaderAction?.Invoke(reader, resultNumber);
+                                    }
+
+                                    resultNumber++;
+                                } while (await reader.NextResultAsync());
+
+                                break;
+                            }
+                            case ExecType.DataSet:
+                            {
+                                using var adapter = new SqlDataAdapter(_command);
+                                adapter.Fill(result.FillDataSetResult);
```

Comment:
Here is sync fill. Maybe it would be good to wrap this line of code to some `Task` to achieve asynchrony ?

Reply:
As far as I know, making something `async` just to `await` it right away is an anti-pattern and should not be used.